src/nodes/chunk_append/planner.c

```diff
@@ -404,11 +404,35 @@ ts_chunk_append_get_scan_plan(Plan *plan)
 			return (Scan *) plan;
 			break;
 		case T_CustomScan:
-			if (castNode(CustomScan, plan)->scan.scanrelid > 0)
+		{
+			CustomScan *custom = castNode(CustomScan, plan);
+			if (custom->scan.scanrelid > 0)
+			{
+				/*
+				 * The custom plan node is a scan itself. This handles the
+				 * DecompressChunk node.
+				 */
 				return (Scan *) plan;
-			else
-				return NULL;
-			break;
+			}
+
+			if (strcmp(custom->methods->CustomName, "VectorAgg") == 0)
```

Comment:
Nit:

In `ts_is_gapfill_path` we also check the name of a TSL node. However, the name of the node is defined in a header file. Can we do the same to be consistent?

Reply:
Sure, I put it into a separate header in the Apache module.